### DISCUSSION TITLE
deployment error [Feb 07]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5097] [Discord Threads]
> 🙋‍♂️ **shikishikichangchang**

**Comment by o1lab** [comment]
> 🙋‍♂️ **shikishikichangchang**  
####   
![image0](https://cdn.discordapp.com/attachments/1061464376982376488/1061464600345853972/image.png)  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### I keep getting this error when deploying. All I did was deploy an express app to vercel.  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### ```let app = require('express')();   
####    
#### const PORT = 8080;   
####    
#### const httpServer = app.listen(PORT, () => {   
####   console.log(`API listening on PORT ${PORT} `);   
#### });   
####    
#### app.use(async function (req, res, next) {   
####   try {   
####     const { Noco } = require('nocodb-daily');   
####     await Noco.init({}, httpServer, app);   
####     next();   
####   } catch (error) {   
####     next(error);   
####   }   
#### });   
####    
#### module.exports = app;   
#### ```  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### Any help with deploying to Vercel?  
#  
> 🙋‍♂️ **wkw**  
#### does it work if u use `nocodb` ? just wanna narrow down   
####    
#### const { Noco } = require('nocodb');  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### Let me try that  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### Doesn't work  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### shows the same error  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### and also  
#  
> 🙋‍♂️ **shikishikichangchang**  
####   
![image0](https://cdn.discordapp.com/attachments/1061464376982376488/1061471180680343612/image.png)  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### What does Noco.init do?  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### It seems like it starts an API but it also serves some frontend Vue pages?  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### Vercel allows devs to serve serverless functions and server pages. Not sure but it seems like Noco.init mixes that two things?  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### I tried this too  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### ```let app = require('express')();   
####    
#### (async () => {   
####   try {   
####     const { Noco } = require('nocodb-daily');   
####     const httpServer = app.listen(process.env.PORT || 8080);   
####     app.use(await Noco.init({}, httpServer, app));   
####   } catch (e) {   
####     console.log(e);   
####   }   
#### })();   
####    
#### module.exports = app;```  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### The error is now: GET https://sportsconnectadmin.vercel.app/dashboard 404  
![image0](https://cdn.discordapp.com/attachments/1061464376982376488/1061479982301454346/image.png)  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### I added a catch all  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### ```app.get('*', (req, res) => {   
####   res.send('All other routes');   
#### });```  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### And now I am seeing  
#  
> 🙋‍♂️ **shikishikichangchang**  
####   
![image0](https://cdn.discordapp.com/attachments/1061464376982376488/1061481465369604236/image.png)  
#  
> 🙋‍♂️ **shikishikichangchang**  
#### Looks like NocoDB's routes are not being seen at all...  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1061464376982376488)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
